File: software/examples/python/example_callback.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-  

HOST = "localhost"
PORT = 4223
UID = "abc" # Change to your UID

from tinkerforge.ip_connection import IPConnection
from tinkerforge.bricklet_color import Color

# Callback function for color callback (in RGBC)
def cb_color(r, g, b, c):
    print('Color(R): ' + str(r))
    print('Color(G): ' + str(g))
    print('Color(B): ' + str(b))
    print('Color(C): ' + str(c))
    print('')


if __name__ == "__main__":
    ipcon = IPConnection() # Create IP connection
    c = Color(UID, ipcon) # Create device object

    ipcon.connect(HOST, PORT) # Connect to brickd
    # Don't use device before ipcon is connected
    # Set Period for color callback to 1s (1000ms)
    # Note: The callback is only called every second if the 
    #       temperature has changed since the last call!
    c.set_color_callback_period(1000)

    # Register temperature callback to function cb_temperature
    c.register_callback(c.CALLBACK_COLOR, cb_color)
    
    raw_input('Press key to exit\n') # Use input() in Python 3
    ipcon.disconnect()

File: software/examples/javascript/ExampleCallback.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'abc'; // Change to your UID

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var c = new Tinkerforge.BrickletColor(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function(error) {
        console.log('Error: '+error);
    }
); // Connect to brickd

// Don't use device before ipcon is connected
ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function(connectReason) {
        // Set Period for color callback to 1s (1000ms)
        // Note: The callback is only called every second if the
        // color has changed since the last call!
        c.setColorCallbackPeriod(1000);
    }
);

// Register temperature callback
c.on(Tinkerforge.BrickletColor.CALLBACK_COLOR,
    // Callback function for color callback
    function(r, g, b, cl) {
        console.log('Color(R): '+r);
        console.log('Color(G): '+g);
        console.log('Color(B): '+b);
        console.log('Color(C): '+cl);
        console.log();
    }
);

console.log("Press any key to exit ...");
process.stdin.on('data',
    function(data) {
        ipcon.disconnect();
        process.exit(0);
    }
);
